### ISSUE TITLE
Fix S4056 and S4058: False positive for string comparisons with StringComparison parameter

### ISSUE THREAD

**Issue by gkello** [SonarSource/sonar-csharp#707]
### Description

Rules S4056 and S4058 have overlapping warnings. For
```c#
str.StartsWith("test")
```
two issues are detcted:
```
Warning	S4056	Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.	
Warning	S4058	Change this call to 'str.StartsWith' to an overload that accepts a 'StringComparison' as a parameter.
```

```c#
str.StartsWith("test", StringComparison.InvariantCulture)
```
has
```
Warning	S4056	Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.
```

and
```C#
str.StartsWith("test", false, CultureInfo.InvariantCulture)
```

```
Warning	S4058	Change this call to 'str.StartsWith' to an overload that accepts a 'StringComparison' as a parameter.
```

The same is true for string.EndsWith and  string.Compare methods

### Repro steps

```C#
        private static bool StartWithRepro(string str)
        {
            if (str.StartsWith("test"))
            {
                return true;
            }
            if (str.StartsWith("test", StringComparison.InvariantCulture))
            {
                return true;
            }
            if (str.StartsWith("test", false, CultureInfo.InvariantCulture))
            {
                return true;
            }
            return false;
        }
```

### Expected behavior

Only 1 issue for method above is generated, suggesting to use choose overload with uses either StringComparison or CultureInfo parameter, similar to
```
Warning	S4058	Change this call to 'str.StartsWith' to an overload that accepts a 'StringComparison' or 'CultureInfo' as a parameter.
```

### Actual behavior

4 issues are generated: 
```
Warning	S4056	Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.
Warning	S4058	Change this call to 'str.StartsWith' to an overload that accepts a 'StringComparison' as a parameter.
Warning	S4056	Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.
Warning	S4058	Change this call to 'str.StartsWith' to an overload that accepts a 'StringComparison' as a parameter.
```
### Related information

* SonarAnalyzer.CSharp 6.3.0.2862
* Visual Studio Professional 2017, version 15.3.1

**Comment by Evangelink** [comment]
Hi @gkello,

Thanks for the feedback. I will make a test to see whether we are missing some specific check or whether the two rules are too much overlapping.

**Comment by michalb-sonar** [comment]
You are right @gkello, using either `CultureInfo` or `StringComparison` overload is ok. Thanks for reporting!
